docs/architecture/app-class-and-template.md

```diff
@@ -0,0 +1,200 @@
+# ApplicationClass and ApplicationTemplate
+
+Managing Applications can be passed on some external system, to remove the burden of integration with Compass from Applications.
+Because of that, we are going to extend our API with IntegrationSystem and ApplicationClass types.
+
+To increase the reusability of the application configuration, we introduce ApplicationTemplate. 
+
+
+## Terminology
+
+- **Application** represents functionality that provides API or Events.
+
+- **ApplicationClass** represents soon-to-be Application that is manged by IntegrationSystem. 
+   
+- **ApplicationTemplate** represents soon-to-be Application with all details already provided (like API and Events) that 
+can be created by providing values for replacing placeholders.
+  
+- **IntegrationSystem** (abbreviation: **IntSys**):
```

Comment:
I removed from terminology `Application` and `Integration System` and linked them from `terminology.md`. I keep `ApplicationClass` and `ApplicationTemplate` here - because these concepts are introduced in this document. In addition to that, I copied those definitions to `terminology.md`.